habitat_sim/agent/controls/default_controls.py

```diff
@@ -36,21 +32,33 @@ def _move_along(scene_node: SceneNode, distance: float, axis: int):
     scene_node.translate_local(ax * distance)
 
 
-def _rotate_local(scene_node: SceneNode, theta: float, axis: int):
-    _rotate_local_fns[axis](scene_node, mn.Deg(theta))
-    scene_node.rotation = scene_node.rotation.normalized()
+def _rotate_local(
+    scene_node: SceneNode, theta: float, axis: int, constraint: Optional[float] = None
+):
+    theta = mn.Deg(theta)
+    if constraint is not None:
```

Comment:
Not sure about python idioms but this might be simpler as:

`if constraint:`

Double negatives are challenging for me to parse.

Reply:
`if <x> is not None` is very idiomatic in python, also, because it is duck typed, `if <x>` when you just want to unwrap an optional is bad as many things besides None are false (including 0!).